### ISSUE TITLE
WFopt optimizable vs optimized

### ISSUE THREAD

**Issue by ye-luo** [QMCPACK/qmcpack#3790]
Currently whether a collection of WF parameters are optimized during WFOpt is controlled at TWF input part via optimize=yes.
This is option is confusing. I'd like to make a distinction of two concepts. Optimizable and optimized.
"optimizable" refers to a piece of WF implements optimizable parameters. This is a knowledge TWF knows at compile time.
"optimized" refers to optimizable parameters actually being optimized during WFOpt. This is a state in WFOpt drivers instead of in the TWF.
I'd like push new implementation following this notion.

**Comment by markdewing** [comment]
Depending on how you're planning to name things in the code - a comment on the language: "optimized" is past tense. "optimizing" might be better.  Even better would be a distinction between these concepts that doesn't depend on the ending to "optimize".

**Comment by ye-luo** [comment]
Agreed. In the implementation, we need to minimize the overlap naming to wipe out any confusion.

**Comment by prckent** [comment]
optimizable parameters , variable parameters , variational parameters , parameters , variables...

**Comment by jtkrogel** [comment]
Internal state is another matter.  I'm all for QMCPACK having more awareness internally about the distinction between parameters that should be optimized now vs. those that support that feature in general.
